src/components/vulnerabilities/OwaspRR.js

```diff
@@ -0,0 +1,505 @@
+        }
+        else {
+          setVulnerability({ ...vulnerability, [name]: value });
+        }
+  }
+
+    const validateVector = (vector) => {
+        const vectorRegex =  new RegExp(/^\(SL:[1,3,5,6,9]\/M:[1,4,9]\/O:[0,4,7,9]\/S:[2,4,5,6,9]\/ED:[1,3,7,9]\/EE:[1,3,5,9]\/A:[1,4,6,9]\/ID:[1,3,8,9]\/LC:[2,6,7,9]\/LI:[1,3,5,7,9]\/LAV:[1,5,7,9]\/LAC:[1,7,9]\/FD:[1,3,7,9]\/RD:[1,4,5,9]\/NC:[2,5,7]\/PV:[3,5,7,9]\)$/);
+        const isValid = vectorRegex.test(vector);
+        setIsValidOwaspVector(isValid);
+        return isValid;
+    }
+
+    const parseVector = (vector) => {
+        return vector.slice(1, -1).split(/\//).map(s => s.split(':'));
+    }
+
+    const computeLikehood = (fields) => {
+        let sum = 0;
+        fields.map(([key, value]) => {
+            if ((key === 'SL') || (key === 'M') || (key === 'O') || (key === 'S')
+                || (key === 'ED') || (key === 'EE') || (key === 'A') || (key === 'ID'))
+            {
+                sum += parseInt(value);
+            }
+        });
+        return sum/8;
+    }
+
+    const computeImpact = (fields) => {
+        let sum = 0;
+        fields.map(([key, value]) => {
+            if ((key === 'LC') || (key === 'LI') || (key === 'LAV') || (key === 'LAC')
+                || (key === 'FD') || (key === 'RD') || (key === 'NC') || (key === 'PV'))
+            {
+                sum += parseInt(value);
+            }
+        });
+        return sum/8;
+    }
+
+    const computeOverall = (impact, likehood) => {
+        if ((impact < 3) && (likehood < 3)) {
+            return 'note';
+        } else if (((impact < 3) && (likehood < 6)) || ((impact < 6) && (likehood < 3))) {
+            return 'low';
+        } else if ((impact < 3) || (likehood < 3) || ((impact < 6) && (likehood < 6))) {
+            return 'medium';
+        } else if ((impact < 6) || (likehood < 6)) {
+            return 'high';
+        } else {
+            return 'critical';
+        }
+    }
+
+    const computeValues = (fields) => {
+        const likehood = computeLikehood(fields);
+        const impact = computeImpact(fields);
+        const overall = computeOverall(impact, likehood);
+        return [impact, likehood, overall];
+    }
+
+
+    const computeOwaspScores = (vector) => {
+        let fields = parseVector(vector);
+        return computeValues(fields);
+    }
+
+    const risk_colors = {
+        'note': { label: 'Note', color: 'var(--blue)' },
+        'low': { label: 'Low', color: 'var(--green)' },
+        'medium': { label: 'Medium', color: '#FF8C00' },
+        'high': { label: 'High', color: 'var(--red)' },
+        'critical': { label: 'Critical', color: 'var(--purple)' }
+    };
+
+    const OwaspChart = () => (
+        <RadarChart outerRadius={180} width={900} height={550} data={chartData}>
+            <PolarGrid />
+            <PolarAngleAxis dataKey="subject"/>
```

Comment:
On dark theme, the names of values on the chart are also black - it is harder to read. I did not find the way how to change the font, yet.

Reply:
Ok, no worries. I'll see if I can fix this after merge. Thanks!

Reply:
Great, thank you!